### ISSUE TITLE
Responsive sizing_mode does not work for plotly backend in HoloViews pane

### ISSUE THREAD

**Issue by MarcSkovMadsen** [holoviz/panel#6026]
I'm on the `main` branch of Panel. I was trying to create a *Chat with hvPlot* application where users can specify the plotting backend they prefer in natural language. Unfortunately the plotly plot does not resize to the width and height of its container.

Its also an issue for me at work as many developers and users prefer the Plotly output. And they also prefer responsive sizing modes such that their apps works on big screens, laptop screens and when doing presentations.

```python
import panel as pn
import hvplot.pandas
from bokeh.sampledata.penguins import data as df

pn.extension("plotly")

plot = df.hvplot.scatter(x='bill_length_mm', y='bill_depth_mm', by='species')

component = pn.Column(
    pn.pane.HoloViews(plot, backend="plotly", sizing_mode="stretch_both"),
    sizing_mode="stretch_both", styles={"background": "salmon"}
)

pn.template.FastListTemplate(
    title="Paint It, Black",
    main=[component],
).servable()
```

![image](https://github.com/holoviz/panel/assets/42288570/1f7848be-48df-4089-898f-1d49bd9123a2)

## Additional Information

The matplotlib and bokeh backends works.

**Comment by MarcSkovMadsen** [comment]
If there is a workaround I would really like to know it. Then I could get the chat app fully working.

**Comment by philippjfr** [comment]
You need to actually activate the plotly backend in hvPlot:

```python
import panel as pn
import hvplot.pandas
from bokeh.sampledata.penguins import data as df

hvplot.extension('plotly')

plot = df.hvplot.scatter(x='bill_length_mm', y='bill_depth_mm', by='species', responsive=True)

component = pn.Column(
    pn.pane.HoloViews(plot, backend="plotly", sizing_mode="stretch_both"),
    sizing_mode="stretch_both", styles={"background": "salmon"}
)

pn.template.FastListTemplate(
    title="Paint It, Black",
    main=[component],
).servable()
```

**Comment by MarcSkovMadsen** [comment]
If you try the example, then something is wrong; The position of the tool icons 

![image](https://github.com/holoviz/panel/assets/42288570/a0a4c3e5-6543-49f0-abe1-67ed8df289fb)

I don't see any errors in the browser console.

I'm on the current `main` branch of Panel using

```bash
plotly==5.18.0 bokeh==3.3.2 holoviews==1.18.1 hvplot==0.9.0
```

**Comment by philippjfr** [comment]
Yep was looking at that this morning. It's missing our patched plotly.css file for some reason.

**Comment by MarcSkovMadsen** [comment]
Does that also solve the "dark" mode issue. In dark mode the tool icons are also dark and cannot be seen.

**Comment by MarcSkovMadsen** [comment]
> You need to actually activate the plotly backend in hvPlot and use the responsive option:
> 
> ```python
> import hvplot
> import panel as pn
> import hvplot.pandas
> from bokeh.sampledata.penguins import data as df
> 
> hvplot.extension('plotly')
> 
> plot = df.hvplot.scatter(x='bill_length_mm', y='bill_depth_mm', by='species', responsive=True)
> 
> component = pn.Column(
>     pn.pane.HoloViews(plot, backend="plotly"), styles={"background": "salmon"}
> )
> 
> pn.template.FastListTemplate(
>     title="Paint It, Black",
>     main=[component],
> ).servable()
> ```

In this example you are cheating :-). You are using hvplot.extension("plotly"). If you are using `pn.extension("plotly")` then its not working.

```python
import hvplot
import panel as pn
import hvplot.pandas
from bokeh.sampledata.penguins import data as df

pn.extension('plotly')

plot = df.hvplot.scatter(x='bill_length_mm', y='bill_depth_mm', responsive=True)

pane = pn.pane.HoloViews(plot, backend="bokeh", sizing_mode="stretch_both")

component = pn.Tabs(
    pane, ("Hello", "Hello"), styles={"background": "salmon"}, sizing_mode="stretch_both"
)

def  handle(_=None):
    component[0]=pn.pane.HoloViews(plot, backend="plotly", sizing_mode="stretch_both")

update = pn.widgets.Button(name="Update", on_click=handle)

pn.template.FastListTemplate(
    title="Paint It, Black",
    sidebar=[update],
    main=[component],
).servable()
```


https://github.com/holoviz/panel/assets/42288570/74d2aa23-3e9f-44fb-b1cd-bd936128fdd4

**Comment by philippjfr** [comment]
That is in no way cheating, it's following what the hvPlot docs tell you to do to use the Plotly backend: https://hvplot.holoviz.org/user_guide/Plotting_with_Plotly.html

Calling`pn.extension('plotly')` has no effect on what hvPlot does and should have no effect on what hvPlot does.

**Comment by MarcSkovMadsen** [comment]
## Workaround

If you use the below

```python
hvplot.extension("plotly")
pn.extension('plotly')
```

Then it works


https://github.com/holoviz/panel/assets/42288570/52ed0099-2774-4e31-a5e0-c63851f7ccdf

**Comment by MarcSkovMadsen** [comment]
> That is in no way cheating, it's following what the hvPlot docs tell you to do to use the Plotly backend: https://hvplot.holoviz.org/user_guide/Plotting_with_Plotly.html
> 
> Calling`pn.extension('plotly')` has no effect on what hvPlot does and should have no effect on what hvPlot does.

Hmmm. So user should combine hvplot and panel extension?

```python
hvplot.extension("plotly")
pn.extension('plotly')
```

**Comment by philippjfr** [comment]
They certainly should have to call the panel extension. What definitely has to happen is that hvPlot has to be told to render using plotly instead of the default which is bokeh, ideally that could be done without calling `hvplot.extension`, i.e. if that's not already possible you should be able to do something like `df.hvplot.scatter(..., backend='plotly')` or similar.

**Comment by MarcSkovMadsen** [comment]
I do need to call `pn.extension("plotly")`. Otherwise I get 

```bash
2023-12-14 13:00:45,048 PlotlyPlot was not imported on instantiation may not render in the served application. Ensure you add the following to the top of your application:

pn.extension('plotly')
```

**Comment by MarcSkovMadsen** [comment]
Is the solution to document in the `HoloViews` reference guide that when using `hvPlot` and the `plotly` backend you will need to add both `hvplot.extension("plotly")` and `pn.extension("plotly")`. It is not obvious. If you come from Streamlit, Gradio etc. backgrounds you don't have to do this kind of thing. Its done for you.

**Comment by philippjfr** [comment]
>I do need to call pn.extension("plotly"). Otherwise I get

Cannot reproduce this.

> It is not obvious. If you come from Streamlit, Gradio etc. backgrounds you don't have to do this kind of thing

I would argue this isn't true at all. hvPlot has an API and you have to correctly use that API to tell it to render with a backend other than the default (i.e. bokeh). The fact that `hvplot.extension` happens to share a name with `pn.extension` does not mean they do the same thing, perhaps for clarity it should be `hvplot.set_backend('plotly')` instead.

**Comment by MarcSkovMadsen** [comment]
But I tell it (dynamically) in the `HoloViews` pane to use the plotly backend. I expected it to be controlled there. But it seems I need to control it twice.

**Comment by philippjfr** [comment]
>But I tell it (dynamically) in the HoloViews pane to use the plotly backend. I expected it to be controlled there. I would expect the HoloViews pane to tell hvPlot/ HoloViews everything it needs to render with the backend I specify.

Unfortunately by that point it's too late, the problem is that hvPlot sets options only for the currently selected backend, which means that if you tell the plot to be `responsive=True` then that only applies to the Bokeh backend by default. This is a limitation of HoloViews and/or hvPlot, and could maybe be addressed there, e.g. we could decide to simply apply options for any plotting library that is loaded at execution time.

**Comment by MarcSkovMadsen** [comment]
This is also demonstrates the problem I have. I would like to enable `bokeh` as my default renderer. But support dynamically changing to plotly and having `stretch_width` working.

![image](https://github.com/holoviz/panel/assets/42288570/94b27703-6edf-4589-a85d-c242b965cd4e)

**Comment by MarcSkovMadsen** [comment]
And setting plotly first and the bokeh is not a workaround.

https://github.com/holoviz/panel/assets/42288570/ba34500d-419f-44e1-9954-e0620398aee2

**Comment by MarcSkovMadsen** [comment]
The reference documentation for the `HoloViews` pane is also much more aligned with what I was trying to do

https://panel.holoviz.org/reference/panes/HoloViews.html#switching-backends

There is an example where its shown (indicated?) you can dynamically switch backends, but the example would never work as it does not add `plotly` neither to the `pn.extension` or some hvplot/ holoviews `extension`. And then there is the problem of responsive resizing too.
